transport/response.go

```diff
@@ -24,6 +24,15 @@ import "io"
 
 // Response is the low level response representation.
 type Response struct {
-	Headers map[string]string
+	Headers Headers
 	Body    io.ReadCloser
 }
+
+// ResponseWriter allows Handlers to write responses in a streaming fashion.
+//
+// Parameters of the response should be modified before calling Write().
+type ResponseWriter interface {
+	io.Writer
+
+	AddHeaders(Headers)
```

Comment:
Let's document that all headers should be written before `Write` is called (that's what `http.ResponseWriter` expects at least).

(discussion for future)
Since YARPC has some notion of headers, we should figure out how those will work from this interface. Specifically, errors are sent in response headers, which can only be sent before anything has been written. If the user returns an error, but the writer has already started, then we can't modify the response headers.